File: 프로그래머스/2/42747. H－Index/H－Index.py
def solution(citations):
    answer = 0
    citations.sort(reverse=True)
    n=len(citations)
    for i in range(n):
        if citations[i] < i+1:
            return i
    return n

# 7 6 6 6 6
# 1 2 3 4 5

# 1 1 1 1
# 1 2 3 4

# 4 3 2 1
# 1 2 3 4

# 6 5 3 1 0
# 1 2 3 4 5

# 6 5 4 1 0
# 1 2 3 4 5

File: 프로그래머스/2/42747. H－Index/H－Index.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> citations) {
    int answer = 0;
    sort(citations.rbegin(), citations.rend());
    for(int i=citations[0]; i>=0; i--){
        int over = 0;
        int under=0;
        for(auto a:citations){
            if (a>=i) over++;
            if (a<=i) under++;
        }
        if(over>=i && under<=i) {
            return i;
        }
    }
    return answer;
}